From 470176a657a40d48b3947ee174d50f8d1655b768 Mon Sep 17 00:00:00 2001
From: Kris Van Hees <kris.van.hees@oracle.com>
Date: Wed, 26 Aug 2020 21:35:39 -0400
Subject: [PATCH] Fix double declaration of yylineno

The yylineno variable was declared in dt_cc.c *and* in dt_lex.c (which
is generated from dt_lex.l).  This poses a compilation issue with some
newer compilers.  The declaration in dt_cc,c has been changed into an
extern declaration, while the one in dt_lex.c is kept as the actual
variable declaration.

Signed-off-by: Kris Van Hees <kris.van.hees@oracle.com>
---
 libdtrace/dt_cc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libdtrace/dt_cc.c b/libdtrace/dt_cc.c
index f29a21f0..f1ac0776 100644
--- a/libdtrace/dt_cc.c
+++ b/libdtrace/dt_cc.c
@@ -91,7 +91,7 @@
 #include <dt_bpf.h>
 #include <bpf_asm.h>
 
-int yylineno;
+extern int yylineno;
 
 #ifdef FIXME
 static const dtrace_diftype_t dt_void_rtype = {

